src/core/lib/event_engine/iomgr_engine.cc

```diff
@@ -50,15 +50,12 @@ struct ClosureData {
   EventEngine::TaskHandle handle;
 };
 
-// Timer limits due to quirks in the iomgr implementation.
-// If deadline <= Now, the callback will be run inline, which can result in lock
-// issues. And absl::InfiniteFuture yields UB.
-absl::Time Clamp(absl::Time when) {
-  absl::Time max = absl::Now() + absl::Hours(8766);
-  absl::Time min = absl::Now() + absl::Milliseconds(2);
-  if (when > max) return max;
-  if (when < min) return min;
-  return when;
+grpc_core::Timestamp ToTimestamp(EventEngine::Duration when) {
```

Comment:
Might be worth putting this in an EventEngine time_util file, along with the EventEngine::Duration -> absl::Timestamp conversion we'll need elsewhere.

Reply:
maybe yeah... let's see if/when we need it